onnxruntime/core/platform/windows/telemetry.cc

```diff
@@ -0,0 +1,192 @@
+
+  process_info_logged = true;
+}
+
+void WindowsTelemetry::LogSessionCreation(uint32_t session_id, int64_t ir_version, const std::string& model_producer_name,
+                                          const std::string& model_producer_version, const std::string& model_domain,
+                                          const std::unordered_map<std::string, int>& domain_to_version_map,
+                                          const std::string& model_graph_name,
+                                          const std::unordered_map<std::string, std::string>& model_metadata,
+                                          const std::string& loadedFrom, const std::vector<std::string>& execution_provider_ids) const {
+  if (global_register_count_ == 0 || enabled_ == false)
+    return;
+
+  // build the strings we need
+
+  std::string domain_to_verison_string;
+  bool first = true;
+  for (auto& i : domain_to_version_map) {
+    if (first) {
+      first = false;
+    } else {
+      domain_to_verison_string += ',';
+    }
+    domain_to_verison_string += i.first;
+    domain_to_verison_string += '=';
+    domain_to_verison_string += std::to_string(i.second);
+  }
+
+  std::string model_metadata_string;
+  first = true;
+  for (auto& i : model_metadata) {
+    if (first) {
+      first = false;
+    } else {
+      model_metadata_string += ',';
+    }
+    model_metadata_string += i.first;
+    model_metadata_string += '=';
+    model_metadata_string += i.second;
+  }
+
+  std::string execution_provider_string;
+  first = true;
+  for (auto& i : execution_provider_ids) {
+    if (first) {
+      first = false;
+    } else {
+      execution_provider_string += ',';
+    }
+    execution_provider_string += i;
+  }
+
+  TraceLoggingWrite(telemetry_provider_handle,
+                    "SessionCreation",
+                    TraceLoggingBool(true, "UTCReplace_AppSessionGuid"),
+                    TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage),
+                    TraceLoggingKeyword(MICROSOFT_KEYWORD_MEASURES),
+                    // Telemetry info
+                    TraceLoggingUInt8(0, "schemaVersion"),
+                    TraceLoggingUInt32(session_id, "sessionId"),
+                    TraceLoggingInt64(ir_version, "irVersion"),
+                    TraceLoggingString(model_producer_name.c_str(), "modelProducerName"),
+                    TraceLoggingString(model_producer_version.c_str(), "modelProducerVersion"),
+                    TraceLoggingString(model_domain.c_str(), "modelDomain"),
+                    TraceLoggingString(domain_to_verison_string.c_str(), "domainToVersionMap"),
+                    TraceLoggingString(model_graph_name.c_str(), "modelGraphName"),
+                    TraceLoggingString(model_metadata_string.c_str(), "modelMetaData"),
+                    TraceLoggingString(loadedFrom.c_str(), "loadedFrom"),
+                    TraceLoggingString(execution_provider_string.c_str(), "executionProviderIds"));
+}
```

Comment:
How are we going to get the adapter Luid? So we can get GPU information.
We used to log it in SessionCreation telemetry.

Reply:
For now we wont have it.    We have options :
1- add an event to the EP's to output what device was selected.
2- do nothing, this event logs which EP was chosen - so we know when GPU was chosen.  and we can use existing telemetry to know which GPU was on the machine.   it will have a gap if the machine has 2 GPU's (what percentage are those?) 

---
In reply to: [336233302](https://github.com/microsoft/onnxruntime/pull/2109#discussion_r336233302) [](ancestors = 336233302)

Reply:
For option 2, are we going to use the device id to find corresponding GPU?
I'm fine with adding option 1 later, since the GPU usage is not high.